deliver/lib/deliver/upload_screenshots.rb

```diff
@@ -9,23 +32,55 @@ def upload(options, screenshots)
       v = app.edit_version
       UI.user_error!("Could not find a version to edit for app '#{app.name}'") unless v
 
-      UI.message("Starting with the upload of screenshots...")
+      @checksums_local = ScreenshotMD5s.new
+      @checksums_remote = ScreenshotMD5s.new
+
+      screenshots.each do |screenshot|
+        md5 = Spaceship::Utilities.md5digest(screenshot.path)
+        @checksums_local.add_md5(screenshot.language, screenshot.device_type, md5)
+      end
+
+      # If we have nothing to delete, there's no point in calling save!
+      nothing_to_delete = true
+
+      v.screenshots.each do |lang, screenshots_for_lang|
+        screenshots_for_lang.each do |current|
+          original_file_name = current.original_file_name
+          matched = Spaceship::UploadFile.deconstruct_upload_filename(original_file_name)
```

Comment:
Are we comparing the file name? If so, that won't work properly. Assume a user uses `snapshot`, the generated file names will always be the same. Instead we should compare the content